webapp/channels/src/components/channel_invite_modal/index.ts

```diff
@@ -45,12 +45,22 @@ function makeMapStateToProps(initialState: GlobalState, initialProps: OwnProps)
     }
 
     return (state: GlobalState, props: OwnProps) => {
+        // Check if this is an ABAC channel to bypass contaminated Redux state
+        const channel = props.channelId ? getChannel(state, props.channelId) : null;
+        const isAbacChannel = Boolean(channel?.policy_enforced);
+
         let profilesNotInCurrentChannel: UserProfile[];
         let profilesInCurrentChannel: UserProfile[];
         let profilesNotInCurrentTeam: UserProfile[];
         let membersInTeam;
 
-        if (props.channelId && props.teamId) {
+        if (isAbacChannel) {
+            // For ABAC channels, return empty arrays to force component to use fresh API data
+            profilesNotInCurrentChannel = [];
+            profilesInCurrentChannel = [];
+            profilesNotInCurrentTeam = [];
```

Comment:
Do we think there will be other places throughout the codebase where we will have to make this exception? Would it be worth checking if the channel is policy_enforced in the selectors themselves?